build/Build.md

```diff
@@ -0,0 +1,190 @@
+
+[2] [WSO2 Private Docker Registry, docker.wso2.com](https://docker.wso2.com/) : <br/>
+* Contains docker images built with the inclusion of regularly updated product packs and is accessible only for its subscribed customers. 
+
+WSO2 recommends using these images in containerized deployments of WSO2 products. This is mainly because WSO2 has aligned its Docker images with its recommended best practices for product deployments.
+
+Following are some of the key features of WSO2 product Docker images, which aligns them with WSO2-recommended best practices for container based product deployments:
+
+* *WSO2 recommends applying changes to the default configurations of a WSO2 product running inside a container via a volume mount. Our docker images support this feature via a dedicated volume mount by the name:* ```wso2-config-volume``` <br/> (Refer to [this article](https://medium.com/faun/wso2-products-dockerized-advanced-usage-part-1-ae255ef02661) for more details)
+
+* *WSO2 recommends introducing additional artifacts / non-configuration files (e.g. third-party libraries, Carbon extensions in the form <br/> of OSGi bundles, Carbon Applications or any security-related artifacts such as Java Keystore files) to a WSO2 product <br/>running inside a container via a volume mount. Our docker images support this feature via a dedicated volume <br/>mount by the name:* ```wso2-artifact-volume``` (Refer to [this article](https://medium.com/faun/wso2-products-dockerized-advanced-usage-part-2-9c851b0db557) for more details)
+
+* *WSO2 recommends introducing patches (or hotfixes) to a WSO2 product running inside a container via a volume mount. <br/>Our docker images support this feature via a dedicated volume mount by the name:* ```wso2-patch-volume``` <br/>(Refer to [this article](https://medium.com/faun/wso2-products-dockerized-advanced-usage-part-2-9c851b0db557) for more details)
+
+* *Supports passing WSO2 product startup options.*
+(Refer to [this article](https://medium.com/faun/wso2-products-dockerized-remote-debug-dockerized-wso2-products-with-intellij-idea-111ca8ae60c) for more details)
+
+WSO2 has followed an array of [best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) and [guidelines](https://github.com/docker-library/official-images#review-guidelines) recommended by the Docker community during the implementation of its product Docker resources.
+
+Some of the best practices and guidelines that are followed for Dockerfiles and entry point scripts are explained below. 
+
+* *Minimized container image size:*
+
+    + *Minimizing the number instructions in the dockerfile that would create sizeable image layers,  leading to reduced image size.* <br/> (Refer to [this section](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers) for advanced details)
```

Comment:
@DilanUA let us start the term `dockerfile` with a capital letter.

Reply:
Applicable everywhere.

Reply:
Done